### ISSUE TITLE
Can't create 32-bit executable with vanilla Windows LDC and "ldc2" executable

### ISSUE THREAD

**Issue by p0nce** [ldc-developers/ldc#1945]
It is supposed to be possible now to create both 32-bit and 64-bit programs from the same LDC 64-bit release, but with a out-of-the-box install here is what happens instead.

Using ldc2-1.1.0-beta6-win64-msvc

Let there be a file `main.d`
```d
void main()
{
}
```

**How to reproduce:**

```
$ ldc2 -march x86 source\main.d                         # does not work!
$ ldc2 -march x86-64 source\main.d                      # OK
$ ldmd2 -m32 source\main.d                              # OK
$ ldmd2 -m64 source\main.d                              # OK
```

**Output**:
```
main.obj : error LNK2019: symbole externe non résolu __Dmodule_ref référencé dans la fonction __D4main16__moduleinfoCtorZ
main.obj : error LNK2019: symbole externe non résolu __d_run_main référencé dans la fonction _main
LINK : error LNK2001: symbole externe non résolu _mainCRTStartup
c:/d/ldc2-1.1.0-beta6-win64-msvc/bin/../lib64\phobos2-ldc.lib : warning LNK4272: type d'ordinateur bibliothèque 'x64' en conflit avec le type d'ordinateur cible 'X86'
c:/d/ldc2-1.1.0-beta6-win64-msvc/bin/../lib64\druntime-ldc.lib : warning LNK4272: type d'ordinateur bibliothèque 'x64' en conflit avec le type d'ordinateur cible 'X86'
main.exe : fatal error LNK1120: 3 externes non résolus
Error: `C:\WINDOWS\system32\cmd.exe /s /c "c:\d\ldc2-1.1.0-beta6-win64-msvc\bin\x86.bat link.exe"` failed with status: 1120
```

**Comment by kinke** [comment]
It's just that the config file lookup (depending on exact target triple `i686-pc-windows-msvc`) seems not to take `-march` into proper consideration. Using `-m32` or the full triple via `-mtriple=i686-pc-windows-msvc` with `ldc2.exe` works as expected.

**Comment by p0nce** [comment]
DUB emits `-march x86` since a long time: https://github.com/dlang/dub/blob/master/source/dub/compilers/ldc.d#L64

**Comment by kinke** [comment]
Alright, the actual issue isn't `-march=x86` (which works), it's DUB using `-march x86`, and that space instead of `=` isn't supported by our manual pre-parsing of the triple for the config file lookup (but by LLVM's command-line parser later)...

**Comment by p0nce** [comment]
Thanks.
Should DUB emit `-march=x86` instead? If this is compatible with older `ldc2` I could change it.

**Comment by kinke** [comment]
Yeah that'd be nice, as it's (and has been) the 'official' way according to the usage message (`-march=<string>`).

**Comment by p0nce** [comment]
So, I was wrong, dub already generates `-march=x86` since a long time.

It still does crash on any x86 Windows dub project when using `--compiler=ldc2`, repro case in attachment.
[repro1945.zip](https://github.com/ldc-developers/ldc/files/683431/repro1945.zip)

**Contents:**
`source/main.d`
```d
void main()
{
}
```

`dub.sdl`
```d
name "toto"
```

**To reproduce**
`$ dub -a x86 --compiler ldc2 -v -f`

It may not be a LDC problem, but it would be nice to understand what happens.

**Comment by kinke** [comment]
Then it's most likely #1941.

**Comment by kinke** [comment]
Yep, fixed by #1942.
